repotracker/repotracker.go

```diff
@@ -589,3 +595,67 @@ func createVersionItems(v *version.Version, ref *model.ProjectRef, project *mode
 	}
 	return nil
 }
+
+func addBuildBreakSubscriptions(v *version.Version, projectRef *model.ProjectRef) error {
+	subscriptionBase := event.Subscription{
+		Type:      event.ResourceTypeVersion,
+		Trigger:   "regression",
+		Selectors: trigger.MakeVersionSelectors(*v),
+	}
+	subscribers := []event.Subscriber{}
+
+	// if the commit author wants build break notifications, subscribe them to this commit
+	if v.AuthorID != "" {
+		subscriber, err := makeBuildBreakSubscriber(v.AuthorID)
+		if err != nil {
+			return err
+		}
+		if subscriber != nil {
+			subscribers = append(subscribers, *subscriber)
+		}
+	}
+	// if the project has build break notifications, subscribe admins if no one subscribed
```

Comment:
Consider the following:
1. Admin enables build break subscriptions.
2. Build breaks. Admin receives notification.
3. User subscribes to build break.
4. Build breaks. User receives notification.

This means that whether an admin receives a notification is governed by whether other users have subscribed to the notification. I think it would make sense to either _never_ notify the admins, or let the admins know why we are notifying them ("build is broken, but committer is not subscribed to the build break notifications").

Also, are users subscribed to build break by default? If not, this would mean that admins have to chase down committers to check a box in Evergreen, which seems user-unfriendly. It seems like this should be opt-out, not opt-in.

Reply:
I agree it's quirky, but I implemented what was described in the design:
`Notify Comitter on Build-break (defined as existing TaskRegression notification, see above)
If no comitter can be found, notify project admins. 
`

It isn't really possible to have build break notifications enabled by default for users, since it requires a setup step of adding your github username. Also the build break checkbox in the project makes it clear that it's in some aspects a fallback: https://github.com/evergreen-ci/evergreen/wiki/Event-Driven-Notifications#build-break-notifications

Reply:
I may make a separate change to make this more usable, but this PR LGTM>